File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.java
class Solution {
    public String[] divideString(String s, int k, char fill) {
        int n = s.length();
        int t = n%k == 0? n/k : n/k + 1;
        String[] ans = new String[t];
        int i = 0, j = 0;
        while( i < n){
            int end = Math.min(n, i+k);
            StringBuilder sb = new StringBuilder(s.substring(i, end));
            while(sb.length() < k){
                sb.append(fill);
            }
            ans[j++] = sb.toString();
            i += k;

        }
        
        return ans;
    }
}

File: 2260-divide-a-string-into-groups-of-size-k/2260-divide-a-string-into-groups-of-size-k.cpp
class Solution {
public:
    vector<string> divideString(string s, int k, char fill) {
        int n = s.size();
        vector<string> ans;
        int i = 0;
        while( i+k <= n){
            string temp = s.substr(i, k);
            ans.push_back(temp);
            i += k;
        }
        if(i < n){
            int sz = k - (n-i);
            string temp = s.substr(i);
            while(sz--){
                temp += fill;
            }
            ans.push_back(temp);
        }
        return ans;
    }
};